content/common-engineering-journal-6.md

```diff
@@ -0,0 +1,76 @@
+
+# Team Health
+
+We have been utilizing the [Retrospective](https://marketplace.visualstudio.com/items?itemName=ms-devlabs.team-retrospectives) extension, by Microsoft devLabs, to effectively manage our iteration retrospectives and monitor our team's health through its team assessment feature. This tool provides a straightforward, quick, and efficient method for these tasks. 
+
+>
+> Team Assessment - _legend for series entries intentionally omitted!_
+>
+> ![Team Assessment](../images/common-engineering-journal-6-1.png)
+>
+
+The visual chart and trendline reveal that **reorganizations**, as indicated by points 1, 2, and 3, along with **leadership changes** shown by point 4, are causing dips in metrics such as clarity, energy, psychological safety, and work-life balance. Conversely, it is evident that the engineering team is becoming increasingly resilient to change. In my opinion, this resilience can be attributed to each team member being empowered to lead within their specific area of expertise and experience.
+
+We encourage you to try the [Retrospective](https://marketplace.visualstudio.com/items?itemName=ms-devlabs.team-retrospectives) extension. In our next discussion, we will explore automation and artificial intelligence, revisiting aspects related to people as we shift our focus towards processes and products.
+
+---
+
+# Automation
+
+The team has achieved notable advancements and enhancements in this area by introducing automation for a variety of tasks. These include maintaining and supporting Azure DevOps operations, as well as upgrading applications to the latest version of .NET 8 using the [.NET Upgrade Assistant](  https://devblogs.microsoft.com/dotnet/upgrade-assistant-general-availability/).
+
+>
+> .NET Upgrade Automation
+>
+> ![Automation Flow](../images/common-engineering-journal-6-2.png)
+>
+
+The **process** involves creating a feature branch in each Azure Repo to identify end-of-life (EOF) .NET code, running the upgrade assistant on this branch, generating a pull request, and creating a technical debt work item to consolidate all actions. While automation significantly reduces the upgrade time from days to seconds, it requires **people** to embrace the automation, action the pull requests, and conduct **automated** testing to validate changes both before and after merging. Our initial focus was overly centered on automation, leading us to realize through experience that **people** involvement is crucial for successful adoption.
+
+Other notable automation processes include **configuration-as-code** validations, such as ensuring that Azure Pipelines are configured according to the last known configuration, and automatically reversing unknown manual changes. 
```

Comment:
Had to pause for a moment here, as I think it should be revert...

Reverse and revert have different meanings:
Reverse: Change something to its opposite direction or state.
[Revert: Return to a previous condition or practice](https://www.difference.wiki/reverse-vs-revert/)

Reply:
Good catch @MerlinDarkstar! Thank you - fixed.